src/visualization/types/Scatter/TimeAutoToggle.tsx

```diff
@@ -0,0 +1,47 @@
+import React, {FC, useState} from 'react'
+import {SlideToggle, ComponentSize} from '@influxdata/clockface'
+import {useVisXDomainSettings} from 'src/visualization/utils/useVisDomainSettings'
+import {Table} from '@influxdata/giraffe'
+interface Props {
+  setDomain: (axis: string, xDomain: [number, number]) => void
+  xDomain: number[] | null
+  table: Table
+}
+const isValidXDomain = xDomain => {
+  return (
+    xDomain &&
```

Comment:
```suggestion
    Array.isArray(xDomain) &&
```

this check is arguably objectively better, since any falsy value will fail the check, and any truthy value that isn't an array will also fail it. it cuts out the very rare situations where this isn't an array, but it's a rare JavaScript object with a `length` property